pkg/machinery/config/types/v1alpha1/v1alpha1_discoveryconfig.go

```diff
@@ -10,12 +10,12 @@ import (
 )
 
 // Enabled implements the config.ClusterDiscovery interface.
-func (c ClusterDiscoveryConfig) Enabled() bool {
-	return c.DiscoveryEnabled
+func (c *ClusterDiscoveryConfig) Enabled() bool {
+	return c.DiscoveryEnabled != nil && *c.DiscoveryEnabled
```

Comment:
Would suggest using a helper function like:

```
func IsTrue(value *bool) bool {
  return value != nil && *value
}
```

Reply:
we could even do a generic function for `*T`... I will try that... or we even had one

Reply:
https://github.com/siderolabs/go-pointer/blob/master/pointer.go#L13

Reply:
updated